File: create_title_pos_E.rb
require 'mysql2'
require 'sqlite3'
require 'matrix'


#######################


class SqlSet
	def initialize()
		@db = SQLite3::Database.new 'word2vec.db'
	end
	def select107
		result = []
		@db.execute('select * from sample_107 ') do |row|
			result.push(row)
		end
		return result
	end
	def select622
		result = []
		@db.execute('select * from sample_622 ') do |row|
			result.push(row)
		end
		return result
	end
end


#######################

class Title
	def initialize()
		@random = Random.new
		@kigo_a = ["!","?","!?","…"]	
		word_a = []
		pos_a = []

		@sql = SqlSet.new
		@result_a = @sql.select622
		@result2_a = @sql.select107
	end
	def create
		tmpTitle = @result_a[@random.rand(0..(@result_a.length - 1))]
		if tmpTitle[11] == 1064 then
			@title = [tmpTitle[2], tmpTitle[3], tmpTitle[4], tmpTitle[5], "…"]
			@titlePos = [tmpTitle[6], tmpTitle[7], tmpTitle[8], tmpTitle[9], "記号"]
		else
			@title = [tmpTitle[2], tmpTitle[3], tmpTitle[4],@kigo_a[@random.rand(0..3)]]
			@titlePos = [tmpTitle[6], tmpTitle[7], tmpTitle[8],"記号"]
		end

		tmpTitle2 = @result2_a[@random.rand(0..(@result2_a.length - 1))]
		case tmpTitle2[11]
		when 537 then
                        @title2 = [tmpTitle2[2], tmpTitle2[3], tmpTitle2[4], "x_doushi"]
                        @titlePos2 = [tmpTitle2[6], tmpTitle2[7], tmpTitle2[8], "x_doushi"]
		when 543 then
                        @title2 = [tmpTitle2[2], tmpTitle2[3], tmpTitle2[4],@kigo_a[@random.rand(0..3)]]
                        @titlePos2 = [tmpTitle2[6], tmpTitle2[7], tmpTitle2[8],"記号"]
		when 539 then
                        @title2 = [tmpTitle2[2], tmpTitle2[3], tmpTitle2[4], tmpTitle2[5] ,"x_mesihi"]
                        @titlePos2 = [tmpTitle2[6], tmpTitle2[7], tmpTitle2[8], tmpTitle[9],"x_meishi"]
		when  536 then
                        @title2 = [tmpTitle2[2], tmpTitle2[3], tmpTitle2[4], tmpTitle2[5], "…"]
                        @titlePos2 = [tmpTitle2[6], tmpTitle2[7], tmpTitle2[8], tmpTitle2[9], "記号"]
		when 535 then
			@title2 = [tmpTitle2[2], tmpTitle2[3], tmpTitle2[4],tmpTitle2[5]]
                        @titlePos2 = [tmpTitle2[6], tmpTitle2[7], tmpTitle2[8],tmpTitle2[9]]	
		when 542 then
                        @title2 = [tmpTitle2[2], tmpTitle2[3], tmpTitle2[4], tmpTitle2[5], "…"]
                        @titlePos2 = [tmpTitle2[6], tmpTitle2[7], tmpTitle2[8], tmpTitle2[9], "記号"]
		when 534 then
                        @title2 = [tmpTitle2[2], tmpTitle2[3], tmpTitle2[4],tmpTitle2[5], @kigo_a[@random.rand(0..3)]]
                        @titlePos2 = [tmpTitle2[6], tmpTitle2[7], tmpTitle2[8],tmpTitle2[9], "記号"]
                when 544 then
                        @title2 = [tmpTitle2[2], tmpTitle2[3], tmpTitle2[4], tmpTitle2[5], "…"]
                        @titlePos2 = [tmpTitle2[6], tmpTitle2[7], tmpTitle2[8], tmpTitle2[9], "記号"]
		when 538 then
                        @title2 = [tmpTitle2[2], tmpTitle2[3], tmpTitle2[4],"x_meishi"]
                        @titlePos2 = [tmpTitle2[6], tmpTitle2[7], tmpTitle2[8],"x_meishi"]
		else
			@title2 = [tmpTitle2[2], tmpTitle2[3], tmpTitle2[4],tmpTitle2[5]]
                        @titlePos2 = [tmpTitle2[6], tmpTitle2[7], tmpTitle2[8],tmpTitle2[9]]
		end
	end
	def title
		output = @title.join("/") + "/ / " + @title2.join("/") 
		return output
	end
	
	def titlePos
		output = @titlePos.join("/") + "/ /" + @titlePos2.join("/")
		return output
	end

	#def totalPageRank
	#	output =  ( Vector.elements(@pagerank)  / @pagerank.length ).to_a 
	#	return output.inject(:+) 
	#end

	def titlePosA
		return @titlePos.concat([" "]).concat(@titlePos2)
	end

	def titleA
		return @title.concat([" "]).concat(@title2)
	end
end

#test = Title.new
#100.times{
#	test.create
#	puts "######"
#	puts test.title
#	puts test.titlePos
#}

File: create_title_pos_E.py
# -*- coding: utf-8 -*-
import sqlite3
import random

class SqlSet:
    dbname = 'word2vec.db'
    conn = sqlite3.connect(dbname)
    c = conn.cursor()

    def select107(self):
        result = []
        for row in self.c.execute('select * from sample_107'):
            result.append(row)
        return result

    def select622(self):
        result = []
        for row in self.c.execute('select * from sample_622'):
            result.append(row)
        return result

    def close(self):
	self.conn.close()

class Title:
    kigo_a = ['!','?','!?','…']
    word_a = []
    pos_a = []

    title = []
    titlePos = []
    title2 = []
    titlePos2 = []

    sql = SqlSet()
    result_a = SqlSet.select622(sql)
    result2_a = SqlSet.select107(sql)
    SqlSet.close(sql)
	
    def create(self):
        tmpTitle = random.sample(self.result_a, 1)

	if tmpTitle[0][11] == 1064:
           self.title =  [tmpTitle[0][2], tmpTitle[0][3], tmpTitle[0][4], tmpTitle[0][5], '…']
           self.titlePos =  [tmpTitle[0][6], tmpTitle[0][7], tmpTitle[0][8], tmpTitle[0][9], '記号']
        else:
           self.title = [tmpTitle[0][2], tmpTitle[0][3], tmpTitle[0][4],random.sample(self.kigo_a,1)]
           self.titlePos = [tmpTitle[0][6], tmpTitle[0][7], tmpTitle[0][8],'記号']
        return self.title

        tmpTitle2 = random.sample(self.result2_a, 1)

        if tmpTitle[0][11] == 537:
            self.title2 = [tmpTitle2[0][2], tmpTitle2[0][3], tmpTitle2[0][4], 'x_doushi']
            self.titlePos2 = [tmpTitle2[0][6], tmpTitle2[0][7], tmpTitle2[0][8], 'x_doushi']
        elif tmpTitle[0][11] == 543:
            self.title2 = [tmpTitle2[0][2], tmpTitle2[0][3], tmpTitle2[0][4],random.sample(self.kigo_a,1) ]
            self.titlePos2 = [tmpTitle2[0][6], tmpTitle2[0][7], tmpTitle2[0][8], '記号']
        elif tmpTitle[0][11] == 539:
            self.title2 = [tmpTitle2[0][2], tmpTitle2[0][3], tmpTitle2[0][4], tmpTitle2[0][5], 'x_mesihi']
            self.titlePos2 = [tmpTitle2[0][6], tmpTitle2[0][7], tmpTitle2[0][8], tmpTitle2[0][9], 'x_mesihi']
        elif tmpTitle[0][11] == 536:
            self.title2 = [tmpTitle2[0][2], tmpTitle2[0][3], tmpTitle2[0][4], tmpTitle2[0][5], '…']
            self.titlePos2 = [tmpTitle2[0][6], tmpTitle2[0][7], tmpTitle2[0][8], tmpTitle2[0][9], '記号']
        elif tmpTitle[0][11] == 535:
            self.title2 = [tmpTitle2[0][2], tmpTitle2[0][3], tmpTitle2[0][4], tmpTitle2[0][5]]
            self.titlePos2 = [tmpTitle2[0][6], tmpTitle2[0][7], tmpTitle2[0][8], tmpTitle2[0][9]]
        elif tmpTitle[0][11] == 542:
            self.title2 = [tmpTitle2[0][2], tmpTitle2[0][3], tmpTitle2[0][4], tmpTitle2[0][5], '…']
            self.titlePos2 = [tmpTitle2[0][6], tmpTitle2[0][7], tmpTitle2[0][8], tmpTitle2[0][9], '記号']
        elif tmpTitle[0][11] == 534:
            self.title2 = [tmpTitle2[0][2], tmpTitle2[0][3], tmpTitle2[0][4], tmpTitle2[0][5], random.sample(self.kigo_a,1)]
            self.titlePos2 = [tmpTitle2[0][6], tmpTitle2[0][7], tmpTitle2[0][8], tmpTitle2[0][9], '記号']
        elif tmpTitle[0][11] == 544:
            self.title2 = [tmpTitle2[0][2], tmpTitle2[0][3], tmpTitle2[0][4], tmpTitle2[0][5], '…']
            self.titlePos2 = [tmpTitle2[0][6], tmpTitle2[0][7], tmpTitle2[0][8], tmpTitle2[0][9], '記号']
        elif tmpTitle[0][11] == 538:
            self.title2 = [tmpTitle2[0][2], tmpTitle2[0][3], tmpTitle2[0][4], 'x_meishi']
            self.titlePos2 = [tmpTitle2[0][6], tmpTitle2[0][7], tmpTitle2[0][8], 'x_meishi']
        else:
            self.title2 = [tmpTitle2[0][2], tmpTitle2[0][3], tmpTitle2[0][4], tmpTitle2[0][5]]
            self.titlePos2 = [tmpTitle2[0][6], tmpTitle2[0][7], tmpTitle2[0][8], tmpTitle2[0][9]]
    def title(self):
	output = '/'.join(self.title) + '/ /' + '/'.join(self.title2)
	return output

    def titlePos(self):
	output = '/'.join(self.titlePos) + '/ /' + '/'.join(self.titlePos2)
	return output

　　def titlePosA(self):
	return self.titlePos.extend([' ']).extend(self.titlePos2)

   def titleA(self):
	return self.title.extend([' ']).extend(self.title2)
